app/components/account/details/account-default.component.ts

```diff
@@ -10,8 +12,21 @@ import { Component } from "@angular/core";
     `,
 })
 
-export class AccountDefaultComponent {
+export class AccountDefaultComponent implements OnInit {
     public static breadcrumb() {
         return { name: "Accounts" };
     }
+
+    constructor(
+        private accountService: AccountService,
+        private router: Router) {
+    }
+
+    public ngOnInit(): void {
+        this.accountService.currentAccountId.subscribe((accountId) => {
```

Comment:
save the subscription and unsubsribe in ngOnDestroy or it will keep calling this everytime you cahnge account